src/approval/simple_multisig.rs

```diff
@@ -0,0 +1,364 @@
+        &self,
+        account_id: &AccountId,
+        _action_request: &ActionRequest<Ac, ApprovalState>,
+    ) -> bool {
+        Au::is_account_authorized(account_id)
+    }
+
+    fn try_approve_with_authorized_account(
+        &self,
+        account_id: AccountId,
+        action_request: &mut ActionRequest<Ac, ApprovalState>,
+    ) -> Result<(), Self::Error> {
+        if !self.is_within_validity_period(&action_request.approval_state) {
+            return Err(ApprovalError::RequestExpired);
+        }
+
+        if action_request
+            .approval_state
+            .approved_by
+            .contains(&account_id)
+        {
+            return Err(ApprovalError::AlreadyApprovedByAccount);
+        }
+
+        action_request.approval_state.approved_by.push(account_id);
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use near_sdk::{
+        borsh::{self, BorshDeserialize, BorshSerialize},
+        env, near_bindgen,
+        test_utils::VMContextBuilder,
+        testing_env, AccountId, BorshStorageKey,
+    };
+
+    use crate::{
+        approval::{
+            simple_multisig::{AccountAuthorizer, ApprovalState, Configuration},
+            ApprovalManager,
+        },
+        near_contract_tools,
+        rbac::Rbac,
+        slot::Slot,
+        Rbac,
+    };
+
+    #[derive(BorshSerialize, BorshDeserialize)]
+    enum Action {
+        SayHello,
+        SayGoodbye,
+    }
+
+    impl crate::approval::Action for Action {
+        type Output = &'static str;
+
+        fn execute(self) -> Self::Output {
+            match self {
+                Self::SayHello => "hello",
+                Self::SayGoodbye => "goodbye",
+            }
+        }
+    }
+
+    #[derive(BorshSerialize, BorshStorageKey)]
+    enum Role {
+        Multisig,
+    }
+
+    #[derive(Rbac, Debug, BorshSerialize, BorshDeserialize)]
+    #[rbac(roles = "Role")]
+    #[near_bindgen]
+    struct Contract {}
+
+    impl ApprovalManager<Action, ApprovalState, Configuration<Self>> for Contract {
+        fn root() -> Slot<()> {
+            Slot::new(b"m")
+        }
+    }
+
+    impl AccountAuthorizer for Contract {
+        fn is_account_authorized(account_id: &near_sdk::AccountId) -> bool {
+            Self::has_role(account_id, &Role::Multisig)
+        }
+    }
+
+    #[near_bindgen]
+    impl Contract {
+        #[init]
+        pub fn new() -> Self {
+            <Self as ApprovalManager<_, _, _>>::init(Configuration::new(2, 10000));
+            Self {}
+        }
+
+        pub fn obtain_multisig_permission(&mut self) {
+            self.add_role(&env::predecessor_account_id(), &Role::Multisig);
+        }
+
+        pub fn create(&mut self, say_hello: bool) -> u32 {
+            let action = if say_hello {
+                Action::SayHello
+            } else {
+                Action::SayGoodbye
+            };
+
+            let request_id = self.create_request(action, ApprovalState::new()).unwrap();
+
+            request_id
```

Comment:
This should just be returning the expression like:
```rust
pub fn create(&mut self, say_hello: bool) -> u32 {
    let action = if say_hello {
        Action::SayHello
    } else {
        Action::SayGoodbye
    };
    self.create_request(action, ApprovalState::new()).unwrap()
}
```

Reply:
@petarvujovic98 Personally, I like it better how @encody has it, actually. It's self-documenting that `create_request` returns an ID.

Reply:
@ryancwalsh I don't have a strong point against it, I just like it when it is more concise but wouldn't have mentioned it if clippy wasn't warning me about it.

Reply:
We should definitely prefer clippy's recommendations. @petarvujovic98 How are you running Clippy? This particular lint is not showing up for me when I just run `cargo clippy`.

Reply:
@encody I am running it in neovim via [rust-tools.nvim](https://github.com/simrat39/rust-tools.nvim) but I don't think I have any special settings. This is what I get:
![Screenshot 2022-09-07 075457](https://user-images.githubusercontent.com/36507050/188799839-a0fbf2ed-8415-46dd-851f-f594f407af5d.png)

I'll try to find if there is anything special about the config that rust-tools.nvim uses.